checks/frozen_deps.go

```diff
@@ -46,14 +46,102 @@ func FrozenDeps(c *checker.CheckRequest) checker.CheckResult {
 		isPackageManagerLockFilePresent,
 		isGitHubActionsWorkflowPinned,
 		isDockerfilePinned,
+		isDockerfileFreeOfInsecureDownloads,
 	)(c)
 }
 
-// TODO(laurent): need to support GCB
+// TODO(laurent): need to support GCB pinning.
+
+func isDockerfileFreeOfInsecureDownloads(c *checker.CheckRequest) checker.CheckResult {
+	return CheckFilesContent(CheckFrozenDeps, "*Dockerfile*", false, c, validateDockerfileDownloads)
+}
+
+func validateDockerfileDownloads(pathfn string, content []byte,
+	logf func(s string, f ...interface{})) (bool, error) {
+	contentReader := strings.NewReader(string(content))
+	res, err := parser.Parse(contentReader)
+	if err != nil {
+		return false, fmt.Errorf("cannot read dockerfile content: %w", err)
+	}
+
+	ret := true
+	files := make(map[string]bool)
+
+	// Walk the Dockerfile's AST.
+	for _, child := range res.AST.Children {
+		cmdType := child.Value
+		// Only look for the 'RUN' command.
+		if cmdType != "run" {
+			continue
+		}
+
+		var valueList []string
+		for n := child.Next; n != nil; n = n.Next {
+			valueList = append(valueList, n.Value)
+		}
+
+		if len(valueList) != 1 {
+			return false, ErrParsingDockerfile
+		}
+
+		// Validate it's not downloading and piping into a shell, like
+		// `curl | bash` (supports `sudo`).
+		r, err := validateCommandIsNotFetchPipeExecute(valueList[0], pathfn, logf)
+		if err != nil {
+			return false, err
+		} else if !r {
```

Comment:
nit: not need for else after return.

Reply:
I thought it was cleaner because it groups the checks and makes it more obvious both checks are for `validateCommandIsNotFetchPipeExecute`. Let me know if you'd rather have me change it anyway.

Reply:
I think it's cleaner to not have the else after if, but your choice since there's no real guideline for Go. You can still group these together by not having a newline in between them.